src/fairseq2/datasets/preference.py

```diff
@@ -215,7 +217,9 @@ def cat_source_and_target(example: dict[str, Any]) -> dict[str, Any]:
             indices_chosen = torch.cat([prompt_indices, target_indices_chosen])
             indices_rejected = torch.cat([prompt_indices, target_indices_rejected])
 
-            prompt_len = len(prompt_indices)
+            prompt_len = 0
+            if mask_src_tokens:
```

Comment:
I think lines 224 and 225 can be slightly optimized by skipping `>=` filtering all together if `mask_src_tokens` is true.